conformance/packages/dns-test/src/name_server.rs

```diff
@@ -350,37 +350,27 @@ impl NameServer<Running> {
         self.container.eavesdrop()
     }
 
-    /// gracefully terminates the name server collecting all logs
-    pub fn terminate(self) -> Result<String> {
-        let pidfile = self.implementation.pidfile(Role::NameServer);
+    pub fn trust_anchor(&self) -> Option<&TrustAnchor> {
+        self.state.trust_anchor.as_ref()
+    }
 
-        // if `terminate` is called right after `start` NSD may not have had the chance to create
-        // the PID file so if it doesn't exist wait for a bit before invoking `kill`
-        let kill = format!(
-            "test -f {pidfile} || sleep 1
-kill -TERM $(cat {pidfile})"
-        );
-        self.container.status_ok(&["sh", "-c", &kill])?;
-        let output = self.state.child.wait()?;
-
-        // the hickory-dns binary does not do signal handling so it won't shut down gracefully; we
-        // will still get some logs so we'll ignore the fact that it fails to shut down ...
-        let is_hickory = matches!(self.implementation, Implementation::Hickory(_));
-        if !is_hickory && !output.status.success() {
-            return Err(
-                format!("could not terminate the `{}` process", self.implementation).into(),
-            );
+    /// Returns the logs collected so far
+    pub fn logs(&self) -> Result<String> {
+        if self.implementation.is_hickory() {
+            self.stdout()
```

Comment:
Could you provide some context here? I'm not sure I understand the conditional. Would `unbound` & `bind` not also write to `stdout`? The previous logic used an `assert` to check `stderr` was empty.

Reply:
unbound and BIND print their logs to stderr, which is the recommended practice. hickory-dns prints its logs to stdout.

the previous logic was using stdout for all implementations because we were checking the stdout of e.g. `docker exec hickory-dns` and docker merges the stdout and stderr of the container process and outputs the combined stream to stdout on the host side. IOW, we were unable to distinguish the container process' stdout from its stderr in the previous implementation but we now can.

Reply:
> which is the recommended practice

Is this UNIX? Or DNS best practice to use stderr? If it’s UNIX let’s file an issue to fix that.

Reply:
I'd say it's a UNIX guideline: errors / warnings / diagnostics / logs go to stderr; output that's meant to be piped between processes goes to stdout.

hickory is a daemon / server so I guess there's more leeway compared to command line tools ... still I think it would be good to align hickory with the other DNS servers